runtime/decimal.c

```diff
@@ -167,6 +168,35 @@ double _bal_decimal_to_float(TaggedPtr tp) {
     return strtod(dblStr, NULL);
 }
 
+DecToIntResult _bal_decimal_to_int(TaggedPtr tp) {
+    decQuad dQuantize;
+    decQuad dZero;
+    decQuadZero(&dZero);
+    decContext cx;
+    initContext(&cx);
+    decQuadQuantize(&dQuantize, taggedToDecQuad(tp), &dZero, &cx);
+    DecToIntResult res;
+    if (cx.status & DEC_Invalid_operation) {
+        // The invalid operation flag is raised 
+        // when 34 digits is not enough to represent quantized decQuad.
+        // This situation can be considered as an overflow scenario.
```

Comment:
Are we sure this handles rounding correctly, particularly at the limits?  Say something in the comment about this.

Reply:
Did you mean handling the rounding when the `DEC_Invalid_operation` is raised?

Reply:
Updated the comment